src/main/java/uk/gov/hmcts/reform/idam/client/models/UserInfo.java

```diff
@@ -6,15 +6,18 @@
 import lombok.Builder;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
+import lombok.NoArgsConstructor;
 
+import java.io.Serializable;
 import java.util.List;
 
 @Getter
 @Builder
 @AllArgsConstructor
+@NoArgsConstructor
```

Comment:
why do you need this?

Reply:
For deserialization. Otherwise i see error like below 
`com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of uk.gov.hmcts.reform.idam.client.models.UserInfo (no Creators, like default construct, exist): cannot deserialize from Object value ...`

Reply:
most likely your app is missing constructor parameters being added in the compile options

`-parameters` it should be on by default in spring boot 2 i think though

Reply:
ok, even `ObjectMapper.readValue(json, UserInfo.class)` giving same error.

Reply:
plenty of teams are using this, try add https://github.com/spring-projects/spring-boot/tree/master/spring-boot-project/spring-boot-starters/spring-boot-starter-json as a dependency

Reply:
either it needs @JsonCreator or default constructor at least.

Reply:
i'd be happier with jsoncreator

Reply:
tried with `spring-boot-starter-json` but same error.